server.go

```diff
@@ -62,28 +62,100 @@ func (r *Runner) EnsureYumUpdated() error {
 }
 
 func (r *Runner) InstallGem(p string) error {
-	err := r.EnsureRubyInstalled()
+	if !r.rvmInstalled {
+		err := r.EnsureRvmInstalled()
+		if err != nil {
+			return err
+		}
+		r.rvmInstalled = true
+	}
+	err := r.Command("gem", "install", p)
 	if err != nil {
 		return err
 	}
-	err = r.Command("gem", "install", p)
+	return nil
+}
+
+func (r *Runner) curlRvm(p string) error {
+	err := r.Command("curl", p)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-func (r *Runner) EnsureRubyInstalled() error {
-	if !r.rubyInstalled {
-		err := r.InstallPackage("epel-release")
-		if err != nil {
-			return err
-		}
-		err = r.InstallPackage("ruby")
-		if err != nil {
-			return err
-		}
-		r.rubyInstalled = true
+func (r *Runner) sourceRvm(p string) error {
+	err := r.Command("source", p)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Runner) installRvm(p string) error {
+	err := r.Command("rvm", p)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Runner) EnsureRvmInstalled() error {
+
+	var rvmPackages string = "epel-release gcc-c++ patch readline readline-devel zlib zlib-devel libffi-devel openssl-devel make bzip2 autoconf automake libtool bison sqlite-devel"
+	var gpgKey string = "-sSL https://rvm.io/mpapis.asc | gpg2 --import -"
```

Comment:
The rest of these variables don't really help readability, because the reader has to keep jumping from where they're used to the definition, and back to where the next variable is used, and back to the definition, and so on.

What we really want is to be able to write something like:

```
r.InstallPackage(rvmDependencies)
r.Command("curl",  "-sSL https://rvm.io/mpapis.asc | gpg2 --import -")
r.Command("curl", "-sSL https://rvm.io/pkuczynski.asc | gpg2 --import -")
...
if r.Error != nil {
	// oh no
}
```

Remember the way errors work on `script` pipelines? Anything that errors, sets the error flag on the pipe. All commands on the pipe check the error flag first, and return immediately if it's set, making themselves a no-op. We could do something similar here. Inside `r.Command()`, if the command returns an error, we could set an error on the Runner object.

Reply:
The weird thing, I originally wrote the code like you suggested but for some reason I kept getting an extra space after the last character. Then I tested it by separating the linux command and the args and it worked fine. I'll try it again and see if I get the same results.